tests/test_visitors/test_ast/test_modules/test_magic_module_functions.py

```diff
@@ -20,13 +20,31 @@ def {0}(name):
 def test_wrong_magic_used(
     assert_errors,
     parse_ast_tree,
-    default_options,
+    options,
+    function_names,
+):
+    """Testing magic methods with logic is restricted."""
+    tree = parse_ast_tree(regular_function.format(function_names))
+    option_values = options(i_control_code=False)
+
+    visitor = MagicModuleFunctionsVisitor(option_values, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [])
+
+
+@pytest.mark.parametrize('function_names', MAGIC_MODULE_NAMES_BLACKLIST)
+def test_right_magic_used(
+    assert_errors,
+    parse_ast_tree,
+    options,
     function_names,
 ):
     """Testing that some magic methods are restricted."""
     tree = parse_ast_tree(regular_function.format(function_names))
+    option_values = options(i_control_code=True)
 
-    visitor = MagicModuleFunctionsVisitor(default_options, tree=tree)
+    visitor = MagicModuleFunctionsVisitor(option_values, tree=tree)
```

Comment:
Let's keep it `default_options`, because this way we also check the default option.